src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -164,3 +183,102 @@ def create(self, stage_name: str, comment: Optional[str] = None) -> SnowflakeCur
+        sorted_file_list = sorted(
+            filtered_file_list, key=lambda f: (path.dirname(f), path.basename(f))
+        )
+
+        sql_variables = self._parse_execute_variables(variables)
+        results = []
+        for file in sorted_file_list:
+            results.append(
+                self._call_execute_immediate(
+                    file=file, variables=sql_variables, on_error=on_error
+                )
+            )
+
+        return results
+
+    def _get_files_list_from_stage(self, stage_path: str) -> List[str]:
+        stage_name = self.get_stage_name_from_path(stage_path)
+        files_list_result = self.list_files(stage_name).fetchall()
+
+        if not files_list_result:
+            raise ClickException(f"No files found on stage '{stage_name}'")
+
+        return [f["name"] for f in files_list_result]
+
+    def _filter_files_list(
+        self, stage_path: str, files_on_stage: List[str]
+    ) -> List[str]:
+        stage_path = self.remove_stage_prefix(stage_path)
+
+        # Exact file path was provided if stage_path in file list
+        if stage_path in files_on_stage:
+            filtered_files = self._filter_supported_files([stage_path])
+            if filtered_files:
+                return filtered_files
+            else:
+                raise ClickException(
+                    "Invalid file extension, only `.sql` files are allowed."
+                )
+
+        # Filter with fnmatch if contains `*` or `?`
+        if glob.has_magic(stage_path):
+            filtered_files = fnmatch.filter(files_on_stage, stage_path)
+        else:
+            # Path to directory was provided
+            filtered_files = fnmatch.filter(files_on_stage, f"{stage_path}*")
+        return self._filter_supported_files(filtered_files)
+
+    @staticmethod
+    def _filter_supported_files(files: List[str]) -> List[str]:
+        return [f for f in files if Path(f).suffix in EXECUTE_SUPPORTED_FILES_FORMATS]
+
+    @staticmethod
+    def _parse_execute_variables(variables: Optional[List[str]]) -> Optional[str]:
+        if not variables:
+            return None
+
+        query_parameters = []
+        for p in variables:
+            if "=" not in p or p.count("=") > 1:
+                raise ClickException(f"Invalid variable: '{p}'")
+
+            key, value = p.split("=")
+            query_parameters.append(f"{key.strip()}=>{value.strip()}")
+        return f" using ({', '.join(query_parameters)})"
+
+    def _call_execute_immediate(
+        self, file: str, variables: Optional[str], on_error: OnErrorType
+    ) -> Dict:
+        try:
+            stage_path_prefixed = self.get_standard_stage_prefix(file)
+            query = (
+                f"execute immediate from {self.quote_stage_name(stage_path_prefixed)}"
+            )
+            if variables:
+                query += variables
+            self._execute_query(query)
+            cli_console.step(f"{file} - SUCCESS")
```

Comment:
I would reverse the order
```suggestion
            cli_console.step(f"SUCCESS - {file}")
```
to achieve more consistent output:
```
SUCCESS - foo.sql
SUCCESS - bar/foo.sql
SUCCESS - foo/bar/arr.sql
```
instead of 
```
foo.sql - SUCCESS
bar/foo.sql - SUCCESS
foo/bar/arr.sql - SUCCESS
```

WDYT?

Reply:
I agree, changed